EXILED/Exiled.Events/Patches/Events/Player/Shooting.cs

```diff
@@ -39,34 +34,59 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
             List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
 
             Label returnLabel = generator.DefineLabel();
-
-            int offset = 1;
-            int index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Stloc_2) + offset;
+            Label continueLabel = generator.DefineLabel();
+
+            /*
+            [] <= Here
+            IL_0078: ldarg.2      // processingMethod
+            IL_0079: ldarg.0      // this
+            IL_007a: ldfld        class ReferenceHub InventorySystem.Items.Firearms.Modules.Misc.ShotBacktrackData::PrimaryTargetHub
+            IL_007f: callvirt     instance void class [mscorlib]System.Action`1<class ReferenceHub>::Invoke(!0/*class ReferenceHub* /)
+             */
+            int hasTargetIndex = newInstructions.FindIndex(instruction => instruction.IsLdarg(2));
+            List<Label> hasTargetLabels = newInstructions[hasTargetIndex].ExtractLabels();
+
+            /*
+            [] <= Here
+            IL_0092: ldarg.2      // processingMethod
+            IL_0093: ldnull       // null
+            IL_0094: callvirt     instance void class [mscorlib]System.Action`1<class ReferenceHub>::Invoke(!0/*class ReferenceHub* /)
+             */
+            int noTargetIndex = newInstructions.FindIndex(hasTargetIndex + 1, instruction => instruction.IsLdarg(2));
+            List<Label> noTargetLabels = newInstructions[noTargetIndex].ExtractLabels();
+
+            ConstructorInfo constructorInfo = Constructor(
+                typeof(ShootingEventArgs),
+                new[] { typeof(InventorySystem.Items.Firearms.Firearm), typeof(ShotBacktrackData).MakeByRefType() });
```

Comment:
instead of manually get a constructor by parameters, you can use
```suggestion
            ConstructorInfo constructorInfo = GetDeclaredConstructors(typeof(ShootingEventArgs))[0];
```

Reply:
oh you removed that, there was a reason?

Reply:
🤨🤨🤨okay